h2c/server.go

```diff
@@ -44,12 +52,46 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	s.trackConn(conn, true)
+	defer s.trackConn(conn, false)
 
 	new(http2.Server).ServeConn(bufConn{conn, bufrw}, &http2.ServeConnOpts{
```

Comment:
Ideally, I think we would want to only ever instantiate a single `http2` server and simply use its shutdown capabilities. Unfortunately, that hasn't been implemented yet: https://go-review.googlesource.com/c/net/+/185059/

Reply:
This will be pretty nice in the future, right now we have no way to actually evict the long running HTTP/2 sessions.